layershellev/src/lib.rs

```diff
@@ -1664,6 +1674,36 @@ impl<T> Dispatch<wp_fractional_scale_v1::WpFractionalScaleV1, ()> for WindowStat
     }
 }
 
+impl<T> Dispatch<wayland_protocols::wp::viewporter::client::wp_viewporter::WpViewporter, ()>
+    for WindowState<T>
+{
+    fn event(
+        _: &mut Self,
+        _: &wayland_protocols::wp::viewporter::client::wp_viewporter::WpViewporter,
+        _: <wayland_protocols::wp::viewporter::client::wp_viewporter::WpViewporter as Proxy>::Event,
+        _: &(),
+        _: &Connection,
+        _: &QueueHandle<Self>,
+    ) {
+        // No events.
+    }
+}
+
+impl<T> Dispatch<wayland_protocols::wp::viewporter::client::wp_viewport::WpViewport, ()>
```

Comment:
if you use delegate_noop, you should not impl it this way, here you impl this trait twice